iib/workers/tasks/opm_operations.py

```diff
@@ -295,196 +295,132 @@ def inner(*args, **kwargs):
+    :param str base_dir: temp directory where opm will be executed.
+    :return: list of package names present in input data.
+    :rtype: [str]
     """
-    from iib.workers.tasks.build import _get_index_database
+    olm_packages = opm_render(input_data, base_dir)
 
-    log.info('Serving data from image %s', from_index)
-    if not is_image_fbc(from_index):
-        db_path = _get_index_database(from_index, base_dir)
-        return opm_registry_serve(db_path=db_path)
+    package_names = [
+        olm_package['name']
+        for olm_package in olm_packages
+        if olm_package['schema'] == 'olm.package'
+    ]
 
-    catalog_dir = get_catalog_dir(from_index, base_dir)
-    return opm_serve(catalog_dir=catalog_dir)
+    return package_names
 
 
-@create_port_filelocks(port_purposes=["opm_port", "opm_pprof_port"])
-def opm_serve(
-    opm_port: int,
-    catalog_dir: str,
-    opm_pprof_port: Optional[int] = None,
-) -> Tuple[int, subprocess.Popen]:
+def _get_olm_package_version(olm_bundle: dict) -> str:
     """
-    Locally start OPM service, which can be communicated with using gRPC queries.
-
-    Due to IIB's paralellism, the service can run multiple times, which could lead to port
-    binding conflicts. Resolution of port conflicts is handled in this function as well.
+    Find and return version of OLM bundle.
 
-    :param int opm_port: OPM port number obtained from create_port_filelock decorator
-    :param int opm_pprof_port: Pprof opm port number obtained from create_port_filelock decorator
-    :param str catalog_dir: path to file-based catalog directory that should be served.
-    :return: tuple containing port number of the running service and the running Popen object.
-    :rtype: (int, Popen)
+    :param BundleImage olm_bundle: olm bundle dictionary
+    :return: OLM bundle version
+    :rtype: str
     """
-    log.info('Serving data from file-based catalog %s', catalog_dir)
+    for property in olm_bundle['properties']:
+        if property['type'] == "olm.package":
+            return property['value']['version']
 
-    cmd = [
-        Opm.opm_version,
-        'serve',
-        catalog_dir,
-        '-p',
-        str(opm_port),
-        '-t',
-        '/dev/null',
-    ]
-
-    if opm_pprof_port:
-        # by default opm uses the 127.0.0.1:6060
-        cmd.extend(["--pprof-addr", f"127.0.0.1:{str(opm_pprof_port)}"])
-
-    cwd = os.path.abspath(os.path.join(catalog_dir, os.path.pardir))
-    result = (
-        opm_port,
-        _serve_cmd_at_port_defaults(cmd, cwd, opm_port),
-    )
-    return result
+    error_msg = "No olm package version found for OLM bundle."
+    log.warning(error_msg)
+    raise IIBError(error_msg)
 
 
-@create_port_filelocks(port_purposes=["opm_port"])
-def opm_registry_serve(
-    opm_port: int,
-    db_path: str,
-) -> Tuple[int, subprocess.Popen]:
+def _get_input_data_path(input_data: str, base_dir: str) -> str:
```

Comment:
same here, its confusing if input is image or location path, how about using something like input_image_or_path /input_image_or_dir?